File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamNoiDung/SanPhamNoiDungForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.SanPhamNoiDung")]
[BasedOnRow(typeof(SanPhamNoiDungRow), CheckNames = true)]
public class SanPhamNoiDungForm
{
    public int SanPhamId { get; set; }
    public string TieuDe { get; set; }
    public int LoaiSection { get; set; }
    public string NoiDungHtml { get; set; }
    public int ThuTu { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamNoiDungForm.ts
import { initFormType, IntegerEditor, PrefixedContext, ServiceLookupEditor, StringEditor } from "@serenity-is/corelib";

export interface SanPhamNoiDungForm {
    SanPhamId: ServiceLookupEditor;
    TieuDe: StringEditor;
    LoaiSection: IntegerEditor;
    NoiDungHtml: StringEditor;
    ThuTu: IntegerEditor;
}

export class SanPhamNoiDungForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.SanPhamNoiDung';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SanPhamNoiDungForm.init) {
            SanPhamNoiDungForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;
            var w2 = IntegerEditor;

            initFormType(SanPhamNoiDungForm, [
                'SanPhamId', w0,
                'TieuDe', w1,
                'LoaiSection', w2,
                'NoiDungHtml', w1,
                'ThuTu', w2
            ]);
        }
    }
}